app/models/sitemap_builder.rb

```diff
@@ -0,0 +1,35 @@
+class SitemapBuilder
+  def first_or_generate
+    Sitemap.first || Sitemap.create(xml: generate_xml(page_slugs + sign_slugs))
+  end
+
+  def update_sitemap
+    Sitemap.first.update(xml: generate_xml(page_slugs + sign_slugs))
+  end
+
+  def generate_xml(slugs)
+    base_url = Rails.application.config.base_url
+
+    builder = Nokogiri::XML::Builder.new do |xml|
+      xml.urlset(xmlns: 'http://www.sitemaps.org/schemas/sitemap/0.9') do
+        slugs.each { |slug| xml.url { xml.loc base_url + slug } }
+      end
+    end
+
+    builder.to_xml
+  end
+
+  private
+
+  def fetch_data_dump
+    Sign.all(xmldump: 1)
```

Comment:
What is this doing? What is this `xmldump` attribute? I sense this could have a better name... 🐘

Reply:
The xmldump attribute is something that we need to pass in a call to freelex to get a collection of information for all of the signs stored in it. Theres a big ol set of methods in the sign model that handle it that I don't really want to touch too much, but maybe this could be called something like pull_sign_data_dump_from_freelex instead?

Reply:
I would call the method `fetch_all_signs` then maybe?